app/workers/evss/disability_compensation_form/job_status.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module EVSS
+  module DisabilityCompensationForm
+    module JobStatus
+      extend ActiveSupport::Concern
+
+      def with_tracking(job_title, saved_claim_id, submission_id)
+        yield
+        job_success(job_title, saved_claim_id, submission_id)
+      rescue StandardError => error
```

Comment:
Looking at this again I think, because of how module inheritance works, we can have base retryable and non-retryable error handlers defined in the module and have the job class call `super` from its override. That way the reported error will be less generic and we can move metrics here too.